lib/features/thread/data/network/thread_api.dart

```diff
@@ -226,57 +241,114 @@ class ThreadAPI {
+      ?? changesEmailMethod.requiredCapabilities;
 
-    final capabilities = getEmailCreated.requiredCapabilities
-      .toCapabilitiesSupportTeamMailboxes(session, accountId);
+    final usedCapabilities = requiredCapabilitiesMethod
+        .toCapabilitiesSupportTeamMailboxes(session, accountId);
 
     final result = await (jmapRequestBuilder
-        ..usings(capabilities))
+        ..usings(usedCapabilities))
       .build()
       .execute();
 
     final resultChanges = result.parse<ChangesEmailResponse>(
       changesEmailInvocation.methodCallId,
       ChangesEmailResponse.deserialize);
 
-    final resultUpdated = result.parse<GetEmailResponse>(
-      getEmailUpdatedInvocation.methodCallId,
-      GetEmailResponse.deserialize);
-
-    final resultCreated = result.parse<GetEmailResponse>(
-      getEmailCreatedInvocation.methodCallId,
-      GetEmailResponse.deserialize);
-
-    final listMailboxIdDestroyed = resultChanges?.destroyed.map((id) => EmailId(id)).toList() ?? <EmailId>[];
+    List<EmailId> destroyedEmailIds = resultChanges
+      ?.destroyed
+      .toEmailIds()
+      .toList() ?? [];
+    State? newStateChanges = resultChanges?.newState;
+    bool hasMoreChanges = resultChanges?.hasMoreChanges ?? false;
+    List<Email>? updatedEmail;
+    List<Email>? createdEmail;
+    State? newStateEmail;
+
+    if (getEmailUpdatedInvocation != null) {
+      final emailResponseUpdated = result.parse<GetEmailResponse>(
+        getEmailUpdatedInvocation.methodCallId,
+        GetEmailResponse.deserialize,
+      );
+      updatedEmail = emailResponseUpdated?.list;
+      newStateEmail = emailResponseUpdated?.state;
+      final notFoundIdsUpdated = emailResponseUpdated?.notFound?.toEmailIds().toList() ?? [];
+      log('ThreadAPI::getChanges:notFoundIdsUpdated = $notFoundIdsUpdated');
+      if (notFoundIdsUpdated.isNotEmpty) {
+        destroyedEmailIds.addAll(notFoundIdsUpdated);
+
+        if (AppConfig.isApiLoggingEnabled) {
+          await LogTracking().addLog(
+            message: 'ThreadAPI::getChanges:notFoundIdsUpdated = ${notFoundIdsUpdated.asListString.toString()}',
+          );
+        }
+      }
+    }
 
+    if (getEmailCreatedInvocation != null) {
+      final emailResponseCreated = result.parse<GetEmailResponse>(
+        getEmailCreatedInvocation.methodCallId,
+        GetEmailResponse.deserialize,
+      );
+      createdEmail = emailResponseCreated?.list;
+      newStateEmail = emailResponseCreated?.state;
+      final notFoundIdsCreated = emailResponseCreated?.notFound?.toEmailIds().toList() ?? [];
+      log('ThreadAPI::getChanges:notFoundIdsCreated = $notFoundIdsCreated');
+      if (notFoundIdsCreated.isNotEmpty) {
+        destroyedEmailIds.addAll(notFoundIdsCreated);
+
+        if (AppConfig.isApiLoggingEnabled) {
+          await LogTracking().addLog(
+            message: 'ThreadAPI::getChanges:notFoundIdsCreated = ${notFoundIdsCreated.asListString.toString()}',
```

Comment:
Should we add `state` alongside with not found list?

Reply:
me too, added